src/python/pants/backend/project_info/rules/source_file_validator.py

```diff
@@ -34,6 +34,8 @@ class DetailLevel(Enum):
 
 
 class ValidateOptions(GoalSubsystem):
+    """Validate sources against regexes."""
```

Comment:
So, I think that we put this here initially because the `lint` goal hadn't been ported to v2. It would be good to port this rather than making it first class, if possible.

Reply:
The reason we do not want to do this is that `validate` works with any arbitrary file, even if there is no owning target, thanks to `SourcesSnapshot`. Meanwhile, `lint` only works with files with owning targets.

Toolchain has come to depend on this ability to do things like validate BUILD files, which usually have no owner.

`lint` is already extremely complex - it would probably be possible, but would be horrible to rewrite it to work with either `SourcesSnapshots` or `Targets`.

--

Instead, I recommend that we rename the goal `validate` to something like `regex-validate`. It's confusing what `validate` vs. `lint` mean. `validate` sounds more generic than it really is.

We would get rid of the `sourcefile-validation` subsystem and move the options into the `GoalSubystem`.

Reply:
In general, I think that we need to bias toward having fewer goals, each of which is fundamentally useful for some purpose.

I'd almost rather include validate as a special case of `lint` then have a separate goal for "per-file lint".

Reply:
> I'd almost rather include validate as a special case of lint then have a separate goal for "per-file lint".

Reply:
All great thoughts, but out of scope for this change, which is literally just about adding a description to an existing goal...

Reply:
I think having this functionality in `lint` makes sense, although it is a good point that it can operate on literally any file, including BUILD files, so I'm not sure how hard that would be.